internal/pkg/instrumentation/manager.go

```diff
@@ -54,6 +58,15 @@ type Manager struct {
 	probeMu         sync.Mutex
 	state           managerState
 	stateMu         sync.RWMutex
+	collectionOpts  *ebpf.CollectionOptions
```

Comment:
This doesn't seem to have a need as a field. It is only ever used in a single call site and always set to the same value. It should just be a local var to https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/2029/files#diff-8008933194d32417cc135227b15916c4d1f575de0abda31bc0545b51d9783bebR435